pkg/services/ngalert/api/api_ruler.go

```diff
@@ -122,6 +126,53 @@ func (srv RulerSrv) RouteDeleteAlertRules(c *models.ReqContext) response.Respons
 	return response.JSON(http.StatusAccepted, util.DynMap{"message": "rules deleted"})
 }
 
+// accessibleRulesFGAC will return the rules that are accessible for the user
+// while checking against FGAC. If none is accessible, an error will be returned.
+func (srv RulerSrv) accessibleRulesFGAC(rules []*ngmodels.AlertRule, hasAccess func(evaluator accesscontrol.Evaluator) bool, logger *log.ConcreteLogger) (map[string]struct{}, error) {
```

Comment:
the function is used in one single place and its interface will not allow any other usage. Therefore, I would recommend make it anonymous and put back where it was.

Reply:
I moved both functions as go-lint was complaining about the cyclomatic complexity of the source function and the cognitive load was super high / hard to grasp.

Reply:
Anyway, they are both gone now.